### ISSUE TITLE
Test Suite Difference

### ISSUE THREAD

**Issue by evinism** [evinism/mistql#94]
The below test case differs between JS and Python. 

```json
{
  "query": "count[4]",
  "data": null,
  "throws": true
}
```

Python seems to return None. Expected behavior is probably that it should throw until we choose to allow it.

Kind of weird though, means `x[(keys x)[0]]` will error for some X. Which come to think of it isn't THAT weird.

**Comment by evinism** [comment]
Piped expressions in object literals also seems to be an issue: `{ spotCount: events | filter type =~ "p[ae]t", dog: "spot" }` is a syntax error in Python, but not in JS

**Comment by evinism** [comment]
Piped expression is solved in `0.4.6`

**Comment by evinism** [comment]
Proposed behavior:

* Indexing via the dot operator is an exact alias of indexing via a string with a single argument

For single arguments:
* Objects can be indexed by strings.
* Arrays and string can be indexed by numbers
* Null can be indexed by both strings and numbers

For multiple arguments:
* Arrays and strings can be indexed only via numbers

Everything else errors with a runtime error.

**Comment by evinism** [comment]
Above has been adopted, implemented, somewhat documented, and released as of 0.4.7
